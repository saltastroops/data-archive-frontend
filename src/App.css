.dimScreen {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  height: 100%;
  width: 100%;
  margin: 0;
  padding: 0;
  background: #000000;
  opacity: 0.5;
  filter: alpha(opacity=30);
  -moz-opacity: 0.3;
  z-index: 101;
}
.signal {
  border: 15px solid #fff;
  border-radius: 30px;
  height: 130px;
  left: 50%;
  margin: -15px 0 0 -15px;
  opacity: 0;
  position: absolute;
  top: 50%;
  width: 130px;

  animation: pulsate 1s ease-out;
  animation-iteration-count: infinite;
}

@keyframes pulsate {
  0% {
    transform: scale(0.1);
    opacity: 0;
  }
  50% {
    opacity: 2;
  }
  100% {
    transform: scale(3);
    opacity: 0;
  }
}

.search-results.table {
  margin-bottom: 20px;
}

.search-results.cell {
  display: flex;
  align-items: center;
  padding-left: 5px;
  white-space: nowrap;
}

.search-results.cell.header {
  background-color: white;
  border-bottom: 1px solid lightgray;
  cursor: pointer;
}

.search-results.cell.header.cart {
  cursor: text;
}

.search-results.cell span {
  overflow: hidden;
  text-overflow: ellipsis;
}

.search-results.observation-header {
  background-color: #f2fffb;
  border-bottom: 1px solid lightgray;
  border-top: 2px solid lightgray;
}

.search-results.file-row-even {
  background-color: rgb(245, 245, 245);
}

.search-results.file-row-odd {
  background-color: white;
}

.ReactVirtualized__Grid {
  outline: none;
}
