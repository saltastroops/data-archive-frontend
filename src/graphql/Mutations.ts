import gql from "graphql-tag";

/**
 * Create a new user.
 *
 * The following arguments must be supplied.
 *
 * affiliation:
 *     The affiliation of the user, such as a university or an institute.
 * email:
 *     Email address. This will be stored in lower case.
 * familyName:
 *     The family name (surname).
 * givenName:
 *     The given name (first name).
 * password:
 *     The password, which must have at least 7 characters.
 * username:
 *     The username, which must not contain upper case letters.
 */
export const SIGNUP_MUTATION = gql`
  mutation SIGNUP_MUTATION(
    $familyName: String!
    $givenName: String!
    $username: String!
    $email: String!
    $affiliation: String!
    $password: String!
  ) {
    signup(
      familyName: $familyName
      givenName: $givenName
      username: $username
      email: $email
      affiliation: $affiliation
      password: $password
    ) {
      id
    }
  }
`;

/**
 * Log the user in.
 *
 * The following arguments must be supplied
 *
 * password:
 *     The password.
 * username:
 *     The uusername.
 */
export const LOGIN_MUTATION = gql`
  mutation LOGIN_MUTATION($username: String!, $password: String!) {
    login(password: $password, username: $username) @client
  }
`;

/**
 * Log the user out.
 */
export const LOGOUT_MUTATION = gql`
  mutation LOGOUT_MUTATION {
    logout @client
  }
`;

/**
 * Update user.
 *
 * The following arguments may be supplied.
 *
 * affiliation:
 *     The affiliation of the user, such as a university or an institute.
 * email:
 *     Email address. This will be stored in lower case.
 * familyName:
 *     The family name (surname).
 * givenName:
 *     The given name (first name).
 * id:
 *     The user ID different from the currently logged in user.
 * newPassword:
 *     The new user password, which must have at lease 7 characters.
 * password:
 *     The password, which must have at least 7 characters and is required.
 * username:
 *     The username, which must not contain upper case letters.
 */
export const UPDATE_USER_MUTATION = gql`
  mutation UPDATE_USER_MUTATION(
    $id: ID
    $familyName: String
    $givenName: String
    $username: String
    $email: String
    $affiliation: String
    $password: String!
    $newPassword: String
  ) {
    updateUser(
      id: $id
      familyName: $familyName
      givenName: $givenName
      username: $username
      email: $email
      affiliation: $affiliation
      password: $password
      newPassword: $newPassword
    ) {
      id
    }
  }
`;
