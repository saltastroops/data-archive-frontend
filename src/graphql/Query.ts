import gql from "graphql-tag";

// TODO UPDATE
/**
 * Retrieving the user information.
 */
const USER_QUERY = gql`
  query USER_QUERY {
    user {
      familyName
      givenName
      roles
    }
  }
`;

export { USER_QUERY };
