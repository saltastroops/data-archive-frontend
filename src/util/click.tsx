import { ReactWrapper } from "enzyme";
import * as React from "react";

/**
 * Call the onClick method of the element wrapped in a wrapper with a mouse
 * click event.
 *
 * This function is intended for use in unit tests only. It is required because
 * the standard  Enzyme simulate does not seem to work for routing (see
 * https://github.com/ReactTraining/react-router/issues/4337).
 *
 * Properties
 * ----------
 * wrapper:
 *     Enzyme wrapper, which should wrap exactly one element.
 */
export default function click(
  wrapper: ReactWrapper<any, React.Component, any>
) {
  wrapper.props().onClick(new MouseEvent("click"));
}
