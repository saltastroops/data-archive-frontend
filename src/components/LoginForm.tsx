import * as _ from "lodash";
import * as React from "react";
import { Mutation } from "react-apollo";
import { Redirect } from "react-router-dom";
import styled from "styled-components";
import { LOGIN_MUTATION } from "../graphql/Mutations";
import { USER_QUERY } from "../graphql/Query";
import InputField from "./basicComponents/InputField";

/**
 * Input for the login form.
 *
 * Properties:
 * -----------
 * password:
 *     The password, which must have at least 7 characters.
 * username:
 *     The username, which must not contain upper case letters.
 * }
 */
interface ILoginFormInput {
  username: string;
  password: string;
}

/**
 * State of the login form.
 *
 * Properties:
 * -----------
 * loggedIn:
 *     Whether the user has been logged in.
 * errors:
 *     Object of error messages.
 * userInput:
 *     Values input by the user.
 */
interface ILoginFormState {
  loggedIn: boolean;
  errors: Partial<ILoginFormInput> & { responseError?: string };
  userInput: ILoginFormInput;
}

/**
 * The cache for the login form.
 *
 * The user input and errors are cached.
 */
export interface ILoginFormCache {
  errors?: Partial<ILoginFormInput> & { responseError?: string };
  userInput?: ILoginFormInput;
}

interface ILoginFormProps {
  cache?: ILoginFormCache;
}

const LoginFormParent = styled.form.attrs({
  className: "column is-4 is-offset-4"
})`
  && {
    padding: 1px;
  }
`;

const Heading = styled.h1.attrs({
  className: "title is-3"
})`
  && {
    text-align: center;
    margin: 20px 0 20px 0;
  }
`;

const ErrorMessage = styled.p.attrs({
  className: "error tile"
})`
  && {
    text-align: left;
    margin: 3px 0 3px 0;
    padding: 2px 0 2px 0;
    background-color: hsl(348, 100%, 61%);
    color: white;
  }
`;

/**
 * Validate the given input and return an object of errors found.
 */
const validateLoginForm = (loginInput: {
  username: string;
  password: string;
}) => {
  // An object to store errors for all fields
  const errors: { username?: string; password?: string } = {};

  // Check if the submitted username is not empty
  if (!loginInput.username) {
    errors.username = `Username cannot be empty.`;
  }

  // Check if the submitted username contains upper case characters
  if (loginInput.username !== loginInput.username.toLowerCase()) {
    errors.username = "Username must be in lowercase";
  }

  // Check if the password is secure enough
  if (loginInput.password.length <= 6) {
    errors.password = "Password should be at least 7 characters long";
  }

  // Return an object consisting of the error messages
  return errors;
};

/**
 * The login form for authenticating the user.
 */
class LoginForm extends React.Component<ILoginFormProps, ILoginFormState> {
  public state = {
    errors: {
      password: "",
      responseError: "",
      username: ""
    },
    loggedIn: false,
    userInput: {
      password: "",
      username: ""
    }
  };

  /**
   * Populate the state from cached values.
   */
  componentDidMount() {
    this.setState(() => (this.props.cache as any) || {});
  }

  handleSubmit = async (e: React.FormEvent<EventTarget>, login: any) => {
    e.preventDefault();

    // Validate the user input fields
    const errors = validateLoginForm(this.state.userInput);

    this.updateState({ errors });

    // Check if there is an error, if there is abort signing in.
    if (errors.password || errors.username) {
      return;
    }

    try {
      const logUserIn = await login({
        variables: { ...this.state.userInput }
      });

      if (logUserIn.data.login) {
        this.updateState({
          errors: {
            password: "",
            responseError: "",
            username: ""
          },
          loggedIn: true,
          userInput: {
            password: "",
            username: ""
          }
        });
      }
    } catch (error) {
      this.updateState({
        errors: {
          ...this.state.errors,
          responseError: error.message
            .replace("Network error: ", "")
            .replace("GraphQL error: ", "")
        }
      });
    }
  };

  /**
   * Update the form content according to the user input.
   */
  onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const name = e.target.name;
    const value = e.target.value;
    // Update the userInput property of the state when input field values change
    this.updateState({
      userInput: {
        ...this.state.userInput,
        [name]: value
      }
    });
  };

  render() {
    const { errors, loggedIn } = this.state;
    const { password, username } = this.state.userInput;

    // Go to the main page after successfully logging in.
    if (loggedIn) {
      return <Redirect to={"/"} />;
    }

    return (
      <Mutation
        mutation={LOGIN_MUTATION}
        refetchQueries={[{ query: USER_QUERY }]}
      >
        {(login: any, { loading, error }: any) => {
          return (
            <LoginFormParent
              data-test={"form"}
              onSubmit={e => this.handleSubmit(e, login)}
            >
              <Heading>Login to the Data Archive</Heading>
              {error ? (
                <ErrorMessage>{errors.responseError}</ErrorMessage>
              ) : null}

              <fieldset disabled={loading} aria-disabled={loading}>
                {/* username */}
                <div className="field">
                  <label className="label">
                    Username
                    <div className={"control is-child"}>
                      <InputField
                        id="username"
                        name="username"
                        value={username || ""}
                        error={errors.username}
                        onChange={this.onInputChange}
                        type="text"
                      />
                    </div>
                  </label>
                </div>

                {/* password */}
                <div className="field">
                  <label className="label">
                    Password
                    <div className={"control is-child"}>
                      <InputField
                        id="password"
                        name="password"
                        value={password || ""}
                        error={errors.password}
                        onChange={this.onInputChange}
                        type="password"
                      />
                    </div>
                  </label>
                </div>

                {/* submit button */}
                <button
                  className="button is-success is-fullwidth is-rounded"
                  data-test="signIn"
                >
                  {loading ? "Signing in..." : "Sign in"}
                </button>
              </fieldset>
            </LoginFormParent>
          );
        }}
      </Mutation>
    );
  }

  /**
   * Update the form state and the cache.
   */
  private updateState = (update: object) => {
    this.setState(
      () => update,
      () => {
        if (this.props.cache) {
          this.props.cache.errors = _.cloneDeep(this.state.errors);
          this.props.cache.userInput = _.cloneDeep(this.state.userInput);
        }
      }
    );
  };
}

export default LoginForm;
