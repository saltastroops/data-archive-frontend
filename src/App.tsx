import * as React from "react";
import { Query } from "react-apollo";
import { Redirect, Route, Switch } from "react-router-dom";
import DataRequestsForm from "./components/dataRequest/DataRequestsForm";
import LoginForm, { ILoginFormCache } from "./components/LoginForm";
import NavigationBar from "./components/NavigationBar";
import RegistrationForm, {
  IRegistrationFormCache
} from "./components/RegistrationForm";
import ISearchFormCache from "./components/searchFormComponents/ISearchFormCache";
import SearchPage from "./components/searchFormComponents/SearchPage";
import UserUpdateForm, {
  IUserUpdateFormCache
} from "./components/UserUpdateForm";
import { USER_QUERY } from "./graphql/Query";

interface IUser {
  familyName: string;
  givenName: string;
  isAdmin: boolean;
}

interface IProtectedRouteProps {
  component: any;
  user: IUser | null | undefined;
  [propName: string]: any;
}

interface ICache {
  loginForm: ILoginFormCache;
  registrationForm: IRegistrationFormCache;
  searchForm: ISearchFormCache;
  userUpdateForm: IUserUpdateFormCache;
}

/**
 * A route which redirects to the login page if the user is not logged in.
 */
function ProtectedRoute({
  component: Component,
  user,
  ...rest
}: IProtectedRouteProps) {
  return (
    <Route
      {...rest}
      render={props =>
        user ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: "/login",
              state: { from: props.location }
            }}
          />
        )
      }
    />
  );
}

interface IAppState {
  user?: IUser;
  screenDimensions: { innerHeight: number; innerWidth: number };
}

/**
 * The data archive.
 */
class App extends React.Component<{}, IAppState> {
  state = {
    screenDimensions: {
      innerHeight: window.innerHeight,
      innerWidth: window.innerWidth
    },
    user: undefined
  };

  private cache: ICache = {
    loginForm: {},
    registrationForm: {},
    searchForm: {},
    userUpdateForm: {}
  };

  public render() {
    return (
      <Query query={USER_QUERY}>
        {({ data, loading }: any) => {
          if (loading) {
            return <p>Loading...</p>;
          }

          const currentUser =
            data && data.user
              ? {
                  familyName: data.user.familyName,
                  givenName: data.user.givenName,
                  isAdmin: data.user.roles.some(
                    (role: string) => role === "ADMIN"
                  )
                }
              : null;

          return (
            <>
              <NavigationBar user={currentUser} />

              <Switch>
                {/* search page */}
                <Route
                  exact={true}
                  path="/"
                  render={() => (
                    <SearchPage
                      cache={this.cache.searchForm}
                      screenDimensions={this.state.screenDimensions}
                    />
                  )}
                />

                {/* registration page */}
                <Route
                  exact={true}
                  path="/register"
                  render={() => (
                    <RegistrationForm cache={this.cache.registrationForm} />
                  )}
                />

                {/* login page */}
                <Route
                  exact={true}
                  path="/login"
                  render={() => <LoginForm cache={this.cache.loginForm} />}
                />

                {/* account details page */}
                <ProtectedRoute
                  user={currentUser}
                  exact={true}
                  path="/account"
                  component={() => (
                    <h1 className="title">
                      {currentUser
                        ? `${currentUser.givenName} ${
                            currentUser.familyName
                          } Account`
                        : "No User"}
                    </h1>
                  )}
                />

                {/* data requests page */}
                <ProtectedRoute
                  user={currentUser}
                  exact={true}
                  path="/data-requests"
                  component={() => <DataRequestsForm />}
                />

                {/* cart page */}
                <Route
                  exact={true}
                  path="/cart"
                  component={() => <h1 className="title">Cart page</h1>}
                />

                {/* admin page */}
                <ProtectedRoute
                  user={currentUser}
                  exact={true}
                  path="/admin"
                  component={() => <h1 className="title">Admin page</h1>}
                />

                {/* update user page */}
                <ProtectedRoute
                  user={currentUser}
                  exact={true}
                  path="/user-update"
                  component={() => (
                    <UserUpdateForm cache={this.cache.userUpdateForm} />
                  )}
                />

                {/* page not found */}
                <Route
                  component={() => <h1 className="title">Page not found</h1>}
                />
              </Switch>
            </>
          );
        }}
      </Query>
    );
  }
}

export default App;
