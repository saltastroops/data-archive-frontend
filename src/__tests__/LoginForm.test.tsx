import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import { MockedProvider } from "react-apollo/test-utils";
import { MemoryRouter } from "react-router";
import wait from "waait";
import App from "../App";
import LoginForm from "../components/LoginForm";
import { LOGIN_MUTATION } from "../graphql/Mutations";
import { USER_QUERY } from "../graphql/Query";
import click from "../util/click";

// Helper function for simulating input field value change.
function inputTyping(wrapper: any, name: string, value: string) {
  wrapper.find(`input[name="${name}"]`).simulate("change", {
    target: { name, value }
  });
}

// Updated userInput state
const updatedState = {
  errors: {
    password: "",
    username: ""
  },
  userInput: {
    password: "securepassword",
    username: "sj"
  }
};

describe("LoginForm Component", () => {
  it("renders the LoginForm component having unpopulated props with no errors", async () => {
    // LoginForm component wrapper.
    const wrapper = mount(
      <MockedProvider>
        <LoginForm />
      </MockedProvider>
    );
    // The actual form.
    const form = wrapper.find('form[data-test="form"]');
    // Expect the snapshot to match the LoginForm component.
    expect(toJson(form)).toMatchSnapshot();
  });

  it("submits valid input successfully", async () => {
    const userDetails = {
      password: "securepassword",
      username: "sj"
    };
    const login = jest.fn();
    const mocks = [
      {
        request: {
          query: LOGIN_MUTATION,
          variables: userDetails
        },
        result: () => {
          login();
          return {
            data: {
              login: true
            }
          };
        }
      },

      {
        request: {
          query: USER_QUERY
        },
        result: {
          data: {
            user: {
              __typename: "User",
              familyName: "test",
              givenName: "test",
              roles: []
            }
          }
        }
      }
    ];

    // LoginForm component wrapper.
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <MemoryRouter initialEntries={["/login"]}>
          <LoginForm />
        </MemoryRouter>
      </MockedProvider>
    );

    // LoginForm component instance
    const instance = wrapper.find("LoginForm").instance() as any;

    // Simulate state change when the username input field value changes
    inputTyping(wrapper, "username", userDetails.username);

    // Simulate state change when the password input field value changes
    inputTyping(wrapper, "password", userDetails.password);

    // Expect the properties username and password of the state to have been updated with the correct value
    expect(instance.state.userInput).toMatchObject({
      password: "securepassword",
      username: "sj"
    });

    const signInButton = wrapper.find('[data-test="signIn"]');

    // Expect the button not to indicate loading
    expect(signInButton.text()).toContain("Sign in");

    // The mutation has not been called (yet)
    expect(login).not.toHaveBeenCalled();

    // Simulate the form submission
    signInButton.simulate("submit");

    // Expect the button to indicate loading
    expect(signInButton.text()).toContain("Signing in");

    // The mutation has been called
    await wait(50); // 50 ms rather than 0 ms to give a potential state change the chance to finish
    wrapper.update();
    expect(login).toHaveBeenCalled();

    // Expect no error message.
    expect(wrapper.find("p").length).toBe(0);
  });

  it("displays an error message if the submitted username is invalid", async () => {
    // LoginForm component wrapper.
    const wrapper = mount(
      <MockedProvider>
        <LoginForm />
      </MockedProvider>
    );

    // LoginForm component instance
    const instance = wrapper.find("LoginForm").instance() as any;

    // Set the state with valid information
    instance.setState({ userInput: updatedState.userInput });

    // Spy on the setState function.
    const setState = jest.spyOn(instance, "setState");

    // Simulate state change when the username input field value changes.
    inputTyping(wrapper, "username", "sJ");

    // Expect setState to have been called
    expect(setState.mock.calls.length).toBe(1);

    const signInButton = wrapper.find('[data-test="signIn"]');

    // Simulate the form submission
    signInButton.simulate("submit");

    // Expect the button to have beeen clicked
    expect(signInButton.text()).toContain("Sign in");

    // Expect an error message.
    expect(wrapper.find("p").length).toBe(1);

    // Expect a meaningful error message
    expect(wrapper.find("p").text()).toContain("Username");
    expect(wrapper.find("p").text()).toContain("lowercase");
  });

  it("displays an error message if the submitted password is invalid", () => {
    // LoginForm component wrapper
    const wrapper = mount(
      <MockedProvider>
        <LoginForm />
      </MockedProvider>
    );

    // LoginForm component instance
    const instance = wrapper.find("LoginForm").instance() as any;

    // Set the state with valid information
    instance.setState({ userInput: updatedState.userInput });

    // Spy on the setState function.
    const setState = jest.spyOn(instance, "setState");

    // Simulate state change when the username input field value changes.
    inputTyping(wrapper, "password", "short");

    // Expect setState to have been called
    expect(setState.mock.calls.length).toBe(1);

    // Simulate the form submission
    const signInButton = wrapper.find('[data-test="signIn"]');
    signInButton.simulate("submit");

    // Expect the button not to indicate loading
    expect(signInButton.text()).toContain("Sign in");

    // Expect an error message.
    expect(wrapper.find("p").length).toBe(1);

    // Expect a meaningful error message
    expect(
      wrapper
        .find("p")
        .first()
        .text()
    ).toContain("Password");
    expect(
      wrapper
        .find("p")
        .first()
        .text()
    ).toContain("7 characters");
  });

  it("displays an error message if the login fails", async () => {
    const userDetails = {
      password: "securepassword",
      username: "sj"
    };
    const login = jest.fn();
    const mocks = [
      {
        error: new Error("The server is having a coffee break!"),
        request: {
          query: LOGIN_MUTATION,
          variables: userDetails
        }
      }
    ];

    // LoginForm component wrapper.
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <MemoryRouter initialEntries={["/login"]}>
          <LoginForm />
        </MemoryRouter>
      </MockedProvider>
    );

    // Simulate state change when the username input field value changes
    inputTyping(wrapper, "username", userDetails.username);

    // Simulate state change when the password input field value changes
    inputTyping(wrapper, "password", userDetails.password);

    // There is no error (yet)
    expect(wrapper.find("p.error").length).toBe(0);

    // Simulate the form submission
    const signInButton = wrapper.find('[data-test="signIn"]');
    signInButton.simulate("submit");

    // There is an error now
    await wait(50); // 50 ms rather than 0 ms to give the state change the chance to finish
    wrapper.update();
    expect(wrapper.find("p.error").length).toBe(1);
    expect(wrapper.find("p.error").text()).toEqual(
      "The server is having a coffee break!"
    );
  });
});
