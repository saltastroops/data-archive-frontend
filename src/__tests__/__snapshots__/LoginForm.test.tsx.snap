// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders the LoginForm component having unpopulated props with no errors 1`] = `
.c0.c0 {
  padding: 1px;
}

.c1.c1 {
  text-align: center;
  margin: 20px 0 20px 0;
}

<form
  className="column is-4 is-offset-4 c0"
  data-test="form"
  onSubmit={[Function]}
>
  <styled.h1>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "title is-3",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c1",
            "rules": Array [
              "
  && {
    text-align: center;
    margin: 20px 0 20px 0;
  }
",
            ],
          },
          "displayName": "styled.h1",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "h1",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h1
        className="title is-3 c1"
      >
        Login to the Data Archive
      </h1>
    </StyledComponent>
  </styled.h1>
  <fieldset
    aria-disabled={false}
    disabled={false}
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Username
        <div
          className="control is-child"
        >
          <InputField
            error=""
            id="username"
            name="username"
            onChange={[Function]}
            type="text"
            value=""
          >
            <div
              className="control"
            >
              <input
                className="input "
                id="username"
                name="username"
                onChange={[Function]}
                type="text"
                value=""
              />
            </div>
          </InputField>
        </div>
      </label>
    </div>
    <div
      className="field"
    >
      <label
        className="label"
      >
        Password
        <div
          className="control is-child"
        >
          <InputField
            error=""
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          >
            <div
              className="control"
            >
              <input
                className="input "
                id="password"
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </div>
          </InputField>
        </div>
      </label>
    </div>
    <button
      className="button is-success is-fullwidth is-rounded"
      data-test="signIn"
    >
      Sign in
    </button>
  </fieldset>
</form>
`;
